src/ui/public/timepicker/kbn_global_timepicker.js

```diff
@@ -25,6 +26,73 @@ UiModules
       $rootScope.toggleRefresh = () => {
         timefilter.refreshInterval.pause = !timefilter.refreshInterval.pause;
       };
+
+      // functions for the time navigation buttons
+
+      // travel forward in time based on the interval between from and to
+      $scope.foward = function () {
+        const time = getFromTo();
+        const diff = time.to.diff(time.from);
+        const origTo = time.to.toISOString();
+
+        time.to.add(diff, 'milliseconds');
+        timefilter.time.from = origTo;
+        timefilter.time.to = time.to.toISOString();
+      };
+
+      // travel backwards in time based on the interval between from and to
+      $scope.back = function () {
+        const time = getFromTo();
+        const diff = time.to.diff(time.from);
+        const origFrom = time.from.toISOString();
+
+        time.from.subtract(diff, 'milliseconds');
+        timefilter.time.to = origFrom;
+        timefilter.time.from = time.from.toISOString();
```

Comment:
minor: consider reordering the to/from assignments for symmetry with all the others.